### PR TITLE
Site Editor: Navigation panel add blue dot to customized templates

### LINKED ISSUE #26451: Indicate customized templates in the site editor navigation
Let's use the blue dot + tooltip pattern to indicate whether a theme template has been customised by a user:

This will align with the latest designs we have for other areas of the UI. E.G. [Template Reverting](https://github.com/WordPress/gutenberg/issues/23421#issuecomment-711960167).

### PR DESCRIPTION
**Pull request by david-szabo97**
## Description
Resolves https://github.com/WordPress/gutenberg/issues/26451

Add a blue dot to customized templates.

## How has this been tested?
- Open site editor
- Open navigation panel
- Navigate to Templates
- Customized templates should have a blue dot next to their name

## Screenshots 
image
image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss
Comment (Addison-Stavlo) [inline packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss:96]: We could go with `$white` since it is defined as this in _colors.scss

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss
Comment (Copons) [inline packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss:132]: How about inline block?

```suggestion
.edit-site-navigation-panel__template-item-customized-dot {
	background: var(--wp-admin-theme-color);
	border-radius: 50%;
	display: inline-block;
	height: $grid-unit-10;
	margin-left: 4px;
	width: $grid-unit-10;
}
```

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/template-navigation-item.js
Comment (Copons) [inline packages/edit-site/src/components/navigation-sidebar/navigation-panel/template-navigation-item.js:51]: I wonder if the dot alone is just too small to be an effective hover area for the tooltip.
Maybe it could make sense to wrap the entire title? 🤔 

Also the tooltip background seems to me === the sidebar background, making it hard to read.

@jameskoster do you have any opinions?

**Review by Copons — COMMENTED** [review]
Works as advertised!
I've left a couple of comments. Nothing major, but I'd like a second opinion from design. 🙂

**Review by Addison-Stavlo — COMMENTED** [review]
This is working pretty well in my testing so far.  Some behavioral nitpicking:  

It feels odd having a tooltip available on hover with no change in cursor to help distinguish that.  Normally it might go from 'defualt' to 'pointer' when mousing over a tooltip, but since we are using 'pointer' in the parent container itself there is nothing telling us that anything special is there.  Combining that with the delay that happens before the tooltip actually pops up, this does feel a bit funky and makes the tooltip feel hard to find and feel a bit more hidden.  Thoughts?

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/template-navigation-item.js
Comment (Addison-Stavlo) [reply packages/edit-site/src/components/navigation-sidebar/navigation-panel/template-navigation-item.js:51]: > Maybe it could make sense to wrap the entire title?

That would probably help with the criticism I had below.  Its hard to get it to pop up or even know its there, especially with the cursor already being a pointer in the parent.

**Review by jameskoster — CHANGES_REQUESTED** [review]
At the moment it seems that from-scratch templates are also being marked with the blue dot. 

In all other applications of the blue dot pattern, it indicates that something has been customised. I don't think it would make sense to deviate that here by highlighting the from-scratch templates, which of course haven't been customised :)

**Comment by Addison-Stavlo** [comment]
> At the moment it seems that from-scratch templates are also being marked with the blue dot.

I wonder if this would be acceptable to address in a follow-up?  It does require some rethinking and updates to how we manage templates and their data to distinguish between these states.  For the first step it could make sense to 'blue dot' any template that is not supplied from a theme (and unedited), as that is a step forward from no blue dots at all.  I would vote to extend this to template parts in this state as well.

Im curious your thoughts on the visible design and interaction @jameskoster ?  As @copons and I noted above - https://github.com/WordPress/gutenberg/pull/26894#discussion_r521450120 - the interaction to get the tooltip can feel a bit odd, given that it is so small, there is a bit of a delay to pop up, and the mouse state doesn't change when going over the blue dot.

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss
Comment (Addison-Stavlo) [inline packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss:103]:
```suggestion
	.components-button {
		display: flex;
		flex-direction: column;
```
It looks like this might have gotten messed up on the master rebase, and its causing it to fail the build.  The `button` is missing a closing brace.  We could either remove `button` now since we are using the selector for `.components-button`, or add the closing brace further below.

Comment (Addison-Stavlo) [reply packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss:103]: Actually seems the nesting ruins the styles, so best just remove the `button {` anyways.

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss
Comment (Addison-Stavlo) [inline packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss:146]: `default` doesn't seem very fitting since it doesn't really denote that something is happening.  While it needs to be different from `pointer` that is enabled in its surrounding element, perhaps something like `help` or `wait` would be more fitting.  The tooltip does take a second to pop up, so something other than `default` would be nice so it would convey to the user to expect 'something' from the interaction.

**Review by Addison-Stavlo — COMMENTED** [review]
> In all other applications of the blue dot pattern, it indicates that something has been customized. I don't think it would make sense to deviate that here by highlighting the from-scratch templates, which of course haven't been customized :)

<s> It looks like this is now fixed after the updates from master. 👍  </s> Or not, im confusing myself in testing.

File: packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss
Comment (jameskoster) [reply packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss:146]: `help` could be good!

Comment (Addison-Stavlo) [inline packages/edit-site/src/components/navigation-sidebar/navigation-panel/style.scss:146]:
```suggestion
	cursor: help;
```

**Comment by david-szabo97** [comment]
Blocked until https://github.com/WordPress/gutenberg/pull/27016 is merged

**Comment by vindl** [comment]
We decided to abandon this approach in https://github.com/WordPress/gutenberg/pull/27041#issuecomment-731074109.

**Pull request closed without merging**
